internal/dispatch/caching/caching.go

```diff
@@ -224,10 +223,6 @@ func (cd *Dispatcher) DispatchExpand(ctx context.Context, req *v1.DispatchExpand
 func (cd *Dispatcher) DispatchReachableResources(req *v1.DispatchReachableResourcesRequest, stream dispatch.ReachableResourcesStream) error {
 	cd.reachableResourcesTotalCounter.Inc()
 
-	if req.OptionalLimit == 0 {
```

Comment:
would this as a bonus effect not cause errors due to incompatible dispatch as in previous code?

Reply:
Yes, that should be the case